packages/std/src/init_handle.rs

```diff
@@ -190,11 +227,70 @@ where
     }
 }
 
+impl<T> TryFrom<Context<T>> for MigrateResponse<T>
+where
+    T: Clone + fmt::Debug + PartialEq + JsonSchema,
+{
+    type Error = StdError;
+
+    fn try_from(ctx: Context<T>) -> Result<Self, Self::Error> {
+        Ok(MigrateResponse {
+            messages: ctx.messages,
+            log: ctx.log,
+            data: ctx.data,
+        })
+    }
+}
+#[derive(Clone, Debug, PartialEq)]
+pub struct Context<T = Empty>
+where
+    T: Clone + fmt::Debug + PartialEq + JsonSchema,
+{
+    messages: Vec<CosmosMsg<T>>,
+    log: Vec<LogAttribute>,
+    data: Option<Binary>,
+}
+
+impl<T> Default for Context<T>
+where
+    T: Clone + fmt::Debug + PartialEq + JsonSchema,
+{
+    fn default() -> Self {
+        Context {
+            messages: vec![],
+            log: vec![],
+            data: None,
+        }
+    }
+}
+
+impl<T> Context<T>
+where
+    T: Clone + fmt::Debug + PartialEq + JsonSchema,
+{
+    pub fn new() -> Self {
+        Context::default()
+    }
+
+    pub fn emit<K: ToString, V: ToString>(&mut self, key: K, value: V) {
```

Comment:
I think `emit` implies that something is actually emitted when this is called. However, you can completely ignore this by discarding the context. Either `add_event` or `add_log` (what is it by the way?) are probably better.

Reply:
`emit` terminology is something that comes from Ethereum. 

I can rename to `add_event`